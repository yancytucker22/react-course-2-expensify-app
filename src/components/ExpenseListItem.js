import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

//import {connect} from 'react-redux';
//import { removeExpense } from '../actions/expenses';

export const ExpenseListItem = ({ id, description, amount, createdAt}) => (
    <div>
    <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
    </Link>     
        <p>
            {numeral(amount/100).format('$0,0.00')}
             - 
            {moment(createdAt).format('MMM Do, YYYY')} 
        </p>
    </div> 
);

// gives us access to dispatch prop
//export default connect()(ExpenseListItem);
export default ExpenseListItem;