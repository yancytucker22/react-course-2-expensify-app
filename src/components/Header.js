import React from 'react';  
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';


// <NavLink to="/create" activeClassName="is-active">Go create</NavLink><br/>
//<NavLink to="/help" activeClassName="is-active">Go help</NavLink><br/>
//<NavLink to="/edit" activeClassName="is-active">Go edit</NavLink>

// export for testing purposes
export const Header= ({ startLogout }) => (
    <header className="header"> 
    <div className="content-container">
      <div className="header__content">
          <Link className="header__title" to="/dashboard" activeclassname="is-active">
            <h1>Expensify</h1>
          </Link>
          <button className="button button--link" onClick={startLogout}>Logout</button>
          </div>
        </div>
    </header>
  );
  

  const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
  });

  // connect to redux . undefined - need no state, 
  export default connect(undefined, mapDispatchToProps)(Header);