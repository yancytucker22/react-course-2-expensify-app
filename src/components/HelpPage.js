import React from 'react';  

const HelpPage = () => (
    <div>
      this is from my helpp page component
    </div>
);

export default HelpPage;