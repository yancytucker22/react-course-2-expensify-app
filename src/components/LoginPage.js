import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

// stateless functional component
export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div class='box-layout__box'>
        <h1 class-name='.box-layout__title'>Expensify</h1>
        <p>It's time to get your expenses under control</p>
        <button className="button" onClick={startLogin}>LogIn with Google</button>

        </div>
    </div>
);   

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
 