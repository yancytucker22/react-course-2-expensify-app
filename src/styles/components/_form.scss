.form {
  display: flex;
  flex-direction: column;
  // target direct descendants of .form
  > * {
      margin-bottom: $m-size;
  }

}
.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}