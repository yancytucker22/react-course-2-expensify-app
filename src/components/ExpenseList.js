import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// stateless funcctional component - named export
export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
        })}
    </div>
);


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };    
};
 
//export default (ExpenseList);
//const ConnectedExpenseList = connect((state) => {
//    return {
//      expenses: state.expenses  
//    };    
//})(ExpenseList);

export default connect(mapStateToProps)(ExpenseList);