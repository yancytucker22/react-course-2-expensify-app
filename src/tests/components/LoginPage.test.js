import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render Login Page', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});


test('should call startLogoin on button click', () => {
    const startLogoin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/> )
    expect(startLogin).toHaveBeenCalled();  
});
