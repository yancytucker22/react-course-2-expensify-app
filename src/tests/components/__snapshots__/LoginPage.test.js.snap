// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Login Page 1`] = `
<div
  className="box-layout"
>
  <div
    class="box-layout__box"
  >
    <h1
      class-name=".box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button"
    >
      LogIn with Google
    </button>
  </div>
</div>
`;
